stdlib/messaging/nats/src/main/java/org/ballerinalang/nats/Utils.java

```diff
@@ -64,11 +65,12 @@ public static Object bindDataToIntendedType(byte[] data, BType intendedType) {
         Object dispatchedData;
         switch (dataParamTypeTag) {
             case TypeTags.STRING_TAG:
-                dispatchedData = new String(data, StandardCharsets.UTF_8);
+                dispatchedData = new BmpStringValue(new String(data, StandardCharsets.UTF_8));
```

Comment:
@manuranga Do we have other construct to create StringValue or is it ok to use this directly?

Reply:
Don't directly use the constructor, please create a java string first and convert to BString using StringUtils.fromString.
BmpStringValue is an implimentation detail that may change. But we can't make it private becase it's used in genrated code.

Reply:
Added this change in d156d8f5812d0781377f0636485f1b95c1772874 but still getting the issue https://github.com/ballerina-platform/ballerina-lang/issues/20306